Sources/Kafka/KafkaConsumer.swift

```diff
@@ -78,60 +61,62 @@ public struct KafkaConsumerEvents: Sendable, AsyncSequence {
 
 /// `AsyncSequence` implementation for handling messages received from the Kafka cluster (``KafkaConsumerMessage``).
 public struct KafkaConsumerMessages: Sendable, AsyncSequence {
-    let stateMachine: NIOLockedValueBox<KafkaConsumer.StateMachine>
+    typealias LockedMachine = NIOLockedValueBox<KafkaConsumer.StateMachine>
+
+    let stateMachine: LockedMachine
+    let pollInterval: Duration
 
     public typealias Element = KafkaConsumerMessage
-    typealias BackPressureStrategy = NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark
-    typealias WrappedSequence = NIOThrowingAsyncSequenceProducer<
-        Result<KafkaConsumerMessage, Error>,
-        Error,
-        BackPressureStrategy,
-        KafkaConsumerMessagesDelegate
-    >
-    let wrappedSequence: WrappedSequence
 
     /// `AsynceIteratorProtocol` implementation for handling messages received from the Kafka cluster (``KafkaConsumerMessage``).
     public struct AsyncIterator: AsyncIteratorProtocol {
-        let stateMachine: NIOLockedValueBox<KafkaConsumer.StateMachine>
-        var wrappedIterator: WrappedSequence.AsyncIterator?
+        // FIXME: there are two possibilities:
+        // 1. Create gcd queue and wait blocking call client.consumerPoll() -> faster reaction on new messages
+        // 2. Sleep in case there are no messages -> easier to implement + no problems with gcd Sendability
```

Comment:
I think both are not great but the good thing is I think we have a solution for this in the future with https://github.com/apple/swift-evolution/blob/main/proposals/0417-task-executor-preference.md we are able to create our own custom executor for a `KafkaConsumer` this can be backed by a p_thread in the end (we can use NIO here) and then just do `withTaskExecutorPreference` in the `next()` call before calling the underlying rd Kafka API. This makes sure we have one thread that we can freely block and we get unblocked as soon as we have a message. It just comes with the overhead of a thread hop. Theoretically we could do this conditionally by trying to poll -> no messages -> executor preference -> blocking poll.

The problem is that this feature is only available on the latest nightly Swift versions. So what I would propose for now is going forward with the `sleep` based implementation and creating an issue for adopting task executor preference here in the future.